### PR TITLE
go: add section about error_internal.go to the errors page

### PR DESCRIPTION
**Pull request by kopiczko**
According to the discussion in SIG operator.

### CODE REVIEW

File: go/errors.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Errors
 
+- [Wrapped Errors](#wrapped-errors)
+- [Custom Errors](#custom-errors)
+- [Matching Errors](#matching-errors)
+- [Handling Third Party Library Errors](#handling-third-party-library-errors)
```
Comment (kopiczko) [inline go/errors.md]: ToC added because the page get long.

**Review by sslavic — APPROVED** [review]
LGTM

File: go/errors.md [hunk 2]
```diff
@@ -207,3 +211,74 @@ if pkg.IsTimeout(err) {
...
+`error_internal.go` file. The rest stays as usual.
+
+Example of real world `error_internal.go` file in `githubclient` package:
+
+```go
+package githubclient
+
+import "github.com/google/go-github/github"
+
+func isGitHub404(err error) bool {
```
Comment (MarcelMue) [inline go/errors.md]: Would this then be a replacement for e.g. the `Prometheus` case we describe above? https://github.com/giantswarm/fmt/pull/74/files#diff-d6aadd2606fee522c4b7c216d7753d33R117

**Review by rossf7 — APPROVED** [review]
LGTM.

I like the approach and using it in `helmclient` as suggested by @tuommaki is a good idea. cc @tomahawk28 @sslavic

**Review by tomahawk28 — APPROVED** [review]

**Review by tuommaki — APPROVED** [review]

**Pull request merged**